llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -3719,6 +3762,8 @@ class VPlan {
   /// Return the live-in VPValue for \p V, if there is one or nullptr otherwise.
   VPValue *getLiveIn(Value *V) const { return Value2VPValue.lookup(V); }
 
+  ArrayRef<VPValue *> getLiveIns() const { return VPLiveInsToFree; }
```

Comment:
The name looks confusingly similar to `getLiveIn` above, but they refer to different things. `getLiveIn` lookups a value from `Value2VPValue` and `getLiveIns` returns the list `VPLiveInsToFree`. Should this be called something like `getLiveInsToFree`?

Also, perhaps worth adding a comment to the function?

Reply:
Thanks, I added a comment to the function. 

At the moment, `VPLiveInsToFree` contains all the VPValues in Value2VPValue (that wasn't always the case). It provides an easy and deterministic way to get all created live-ins. The comment for `VPLiveInsToFree` should already be clear as to what it contains, but the name may be a bit confusing, especially with the new use. I updated it to `VPLiveIns`.

Reply:
@david-arm WDYT about the latest naming? I'd be planning to land this soon and potentially iterate on the way live-ins are stored separately if needed.